stress/shell/src/main/java/alluxio/stress/cli/StressMasterBench.java

```diff
@@ -473,47 +555,206 @@ private void runInternal() throws Exception {
             "Thread missed barrier. Increase the start delay. start: %d current: %d",
             mContext.getStartMs(), CommonUtils.getCurrentMs()));
       }
+//      Thread.currentThread().getId()
       CommonUtils.sleepMs(waitMs);
+      Random random = new Random();
+      long totalCount;
+      while (true) {
+        if (Thread.currentThread().isInterrupted()) {
+          break;
+        }
+        if (!useStopCount && CommonUtils.getCurrentMs() >= mContext.getEndMs()) {
+          break;
+        }
+        totalCount = mContext.getTotalCounter().getAndIncrement();
+        if (useStopCount && totalCount >= mParameters.mStopCount) {
+          break;
+        }
+        //select an operation
+        double r = random.nextDouble();
+        int operationIndex = 0;
+        for (int i = 0; i < mParameters.mOperations.length; i++) {
+          if (ratioArray[i] >= r) {
+            operationIndex = i;
+            break;
+          }
+        }
 
-      long localCounter;
+        long operationCount;
+        operationCount = mContext.getOperationCounter(operationIndex).getAndIncrement();
+        rl.acquire();
+        long startNs = System.nanoTime();
+        applyOperation(operationCount, mParameters.mOperations[operationIndex], operationIndex);
+        long endNs = System.nanoTime();
+
+        long currentMs = CommonUtils.getCurrentMs();
+        // Start recording after the warmup
+        if (currentMs > recordMs) {
+          mResult.incrementNumSuccess(1, operationIndex);
+
+          // record response times
+          long responseTimeNs = endNs - startNs;
+          mResponseTimeNs[operationIndex].recordValue(responseTimeNs);
+
+          // track max response time
+          long[] maxResponseTimeNs = mResult.getStatistics(operationIndex).mMaxResponseTimeNs;
+          int bucket =
+              Math.min(maxResponseTimeNs.length - 1, (int) ((currentMs - recordMs) / bucketSize));
+          if (responseTimeNs > maxResponseTimeNs[bucket]) {
+            maxResponseTimeNs[bucket] = responseTimeNs;
+          }
+        }
+      }
+    }
+
+    @SuppressFBWarnings("BC_UNCONFIRMED_CAST")
+    private void runInternalBasedThreadsRatio() throws Exception {
+      String[] components = Thread.currentThread().getName().split("-");
+      double ratioSum = 0;
+      double last = 0;
+      double[] ratioArray = new double[mParameters.mOperations.length];
+      for (int i = 0; i < mParameters.mOperations.length; i++) {
+        ratioSum += mParameters.mThreadsRatio[i];
+      }
+
+      for (int i = 0; i < mParameters.mOperations.length; i++) {
+        ratioArray[i] = last + (mParameters.mThreadsRatio[i] / ratioSum) * mParameters.mThreads;
+        last = ratioArray[i];
+      }
+
+      int threadID = Integer.parseInt(components[components.length - 1 ]);
+      int operationIndex = 0;
```

Comment:
创建一个新的线程类，可以在创建线程的时候指定每个线程应该做的操作
```suggestion
      int operationIndex = 0;p re a
```